src/python/pants/backend/go/target_type_rules.py

```diff
@@ -355,7 +360,12 @@ async def generate_targets_from_go_mod(
     go_mod_snapshot = await Get(Snapshot, Digest, go_mod_info.digest)
     all_packages = await Get(
         AllThirdPartyPackages,
-        AllThirdPartyPackagesRequest(go_mod_info.digest, go_mod_info.mod_path),
+        AllThirdPartyPackagesRequest(
+            go_mod_info.digest,
+            go_mod_info.mod_path,
+            # TODO: Figure out how to avoid rule graph cycle and properly extract these options.
```

Comment:
Need advice on how to avoid rule graph cycle in this rule if this rule calls back into the `GoBuildOptionsFromTargetRequest -> GoBuildOptions` rule.

Reply:
cc @stuhood @Eric-Arellano